src/app/_index.js

```diff
@@ -219,61 +224,105 @@ const beforeLoading = (backupOurs, uploadOurs, fileId, offlineFile, email, userI
   return Promise.resolve(false)
 }
 
+function handleNoFileId(fileId, filePath) {
+  const errorObject = new Error('Could not open cloud file.')
+  rollbar.error(
+    `Attempted to open ${filePath} as a cloud file, but it's not a cloud file.  We think it's id is ${fileId} based on that name.`,
+    errorObject
+  )
+  logger.error(
+    `Attempted to open ${filePath} as a cloud file, but it's not a cloud file.  We think it's id is ${fileId} based on that name.`,
+    errorObject
+  )
+  dialog.showErrorBox(t('Error'), t('There was an error doing that. Try again'))
+  return Promise.reject(new Error(`Cannot open file with id: ${fileId} from filePath: ${filePath}`))
+}
+
+const renderCloudFile = (forceDashboard) => () => {
+  render(
+    <Provider store={store}>
+      <Listener />
+      <Renamer />
+      <App forceProjectDashboard={forceDashboard} />
+    </Provider>,
+    root
+  )
+}
+
+function handleNoUser(filePath) {
+  logger.warn(`Booting a cloud file at ${filePath} but the user isn't logged in.`)
+  renderCloudFile()()
+  return
+}
+
+function handleNoUserId(filePath) {
+  const errorMessage = `Tried to boot plottr cloud file (${filePath}) without a user id.`
+  rollbar.error(errorMessage)
+  return Promise.reject(new Error(errorMessage))
+}
+
+const handleEroneousUserStates = (filePath) => (user) => {
+  // I'm not sure why this branch came about.  It doesn't look right at all.
+  if (!user) {
```

Comment:
@cameronsutter, I think you might've added this branch a while ago.  If you did, could you explain why we render the app anyway?